### ISSUE TITLE
proc typed args weren't bound correctly

### ISSUE THREAD

**Issue by glyh** [oilshell/oil#1892]
Oils 0.21.0		https://www.oilshell.org/

git commit = cc2f68315a18a69be9f8c94bc97c655f5d50bdc2
```
proc DUMP(...a;...b;...c; d) {
  = a
  = b
  = c
  = d
}

DUMP a b c ('d', 'e', 'f', h = "i", j = "k") {
  echo 0
}
```

```
(List 0x22c5)   ["a","b","c"]
(List 0x22ad)   ["e","f",<Block 0x22f1>]
(Dict 0x22a1)   {"h":"i","j":"k"}
(Str)   "d"
```
Should be:
```
["a", "b", "c"]
["d", "e"]
{h: "i", j: "k"}
<Block>
```

**Comment by andychu** [comment]
I believe this is the same as #1849

**Comment by glyh** [comment]
Does my PR solved #1849 as well then?

**Comment by andychu** [comment]
Yes this is the same bug as #1849
